sound/soc/sof/imx/imx8.c

```diff
@@ -126,7 +126,7 @@ static struct imx_dsp_ops dsp_ops = {
 
 static int imx8_send_msg(struct snd_sof_dev *sdev, struct snd_sof_ipc_msg *msg)
 {
-	struct imx8_priv *priv = (struct imx8_priv *)sdev->private;
+	struct imx8_priv *priv = (struct imx8_priv *)sdev->pdata->hw_pdata;
```

Comment:
superfluous type-casts here and everywhere below, you have a good chance to remove them now :-)